pipeline/ingest/loaders/vincent.py

```diff
@@ -51,23 +68,75 @@ def load_sessions(self, subject_name):
             with open(sess.absolute()) as f:
                 sessinfo = json.load(f)
             sess_datetime = datetime.strptime(sessinfo['date'], '%d-%b-%Y %H:%M:%S')
+            # if needed, short form date can be retrieved here. Else, get it from sess_datetime
+            # sess_shortdate = sessinfo['shortDate']
             # find the basename of the other related files for this session
+            sess_basename = sessinfo['baseName']
             data_dir = sess.parent
             # find all associated files
             session_files = [sess.relative_to(self.root_data_dir)
-                             for sess in data_dir.glob(f'{data_dir.name}*')]
+                             for sess in data_dir.glob(f'{sess_basename}*')]
 
             yield {'subject_id': subject_name,
                    'session_date': sess_datetime.date(),
                    'session_time': sess_datetime.time(),
+                   'session_basename': sess_basename,
                    'session_files': session_files,
                    'username': self.config['custom']['username'],
                    'rig': self.config['custom']['rig']}
 
-    def load_behavior(self):
-        pass
+    def load_behavior(self, session_dir, subject_name, session_basename):
+        # TODO: return data entries for tables
+        #   Task            task
+        #   Photostim       return all info available,
+        #           PhotostimLocation Location can be provided later
+        #   SessionTrial    trial number / start_time stop_time
+        #   PhotostimTrial  trial number for trials that contain photostim
+        #   PhotostimEvent  PhotostimTrial photostim_event_id  Photostim photostim_event_time power
+        #   Project         project_name
+
+        # ---- get task type from the session file ----
+        session_info_file = list(session_dir.glob(f'{session_basename}*.json'))
+        with open(session_info_file[0]) as f:
+            sessinfo = json.load(f)
+        task = sessinfo.get('task', 'hf wheel') # if task not specified, default to head-fixed wheel running
```

Comment:
should be: `sessioninfo.get('task', self.default_task)`, as `default_task` is defined above, instead of hardcoding `hf wheel` here